client/consensus/grandpa/lib.go

```diff
@@ -17,9 +19,17 @@ type Address string
 
 // Authority represents a grandpa authority
 type Authority struct {
-	Key    PublicKey
+	Key    ed25519.PublicKey
 	Weight uint64
 }
 
 // AuthorityList A list of Grandpa authorities with associated weights.
 type AuthorityList []Authority
+
+// NewAuthoritySetStruct A new authority set along with the canonical block it changed at.
+type NewAuthoritySetStruct[H comparable, N constraints.Unsigned] struct {
+	CanonNumber N
+	CanonHash   H
+	SetId       N
+	Authorities AuthorityList
+}
```

Comment:
why do we need these changes?

Reply:
It is based on this structure in substrate https://github.com/paritytech/substrate/blob/19971bd3eafa6394d918030f4142f85ea54404c0/client/consensus/grandpa/src/lib.rs#L371

I added "struct" to the end because we already have a function called NewAuthoritySet